drivers/flash/Kconfig.simulator

```diff
@@ -26,6 +26,14 @@ config FLASH_SIMULATOR_DOUBLE_WRITES
 	 If selected, writing to a non-erased program unit will succeed, otherwise, it will return an error.
 	 Keep in mind that write operations can only pull bits to zero, regardless.
 
+config FLASH_SIMULATOR_NOALLOC
+	bool "No buffer allocated, reads, writes and erase are simulated"
```

Comment:
as I read the code, those are not simulated, they are stubbed.

If I do multiple writes, I cannot readback later what I wrote from what I see.

This is a limitation that should be clearer, and because of this I would like a better term than `NOALLOC`, cause it's not just not allocating memory but still functioning in similar way, it's changing behavior also.